addons/web/static/src/js/services/data_manager.js

```diff
@@ -406,8 +406,12 @@ return core.Class.extend({
                 if (fieldsInfo[node.attrs.name].fieldDependencies) {
                     var deps = fieldsInfo[node.attrs.name].fieldDependencies;
                     for (var dependency_name in deps) {
+                        var dependency_dict = {name: dependency_name, type: deps[dependency_name].type};
                         if (!(dependency_name in fieldsInfo)) {
-                            fieldsInfo[dependency_name] = {'name': dependency_name, 'type': deps[dependency_name].type, 'options':  deps[dependency_name].options || {}};
+                            fieldsInfo[dependency_name] = _.extend(dependency_dict, {options: deps[dependency_name].options || {}});
```

Comment:
_.extend({}, dependency_dict, ...) otherwise, you're updating dependency_dict in place